src/main/scala/cromwell/engine/SymbolStore.scala

```diff
@@ -26,9 +26,47 @@ class SymbolStore(binding: WdlBinding, inputs: Map[FullyQualifiedName, WdlValue]
     assignSymbolStoreEntry(fullyQualifiedName, value, input = true)
   }
 
+  for {
+    workflow <- binding.workflows
+    call <- workflow.calls
+    (k, v) <- call.inputMappings
+  } yield assignSymbolStoreEntry(s"${call.fullyQualifiedName}.$k", v, input = true)
+
+  private def callEntries(call: Call): Set[SymbolStoreEntry] =
+    store.filter {entry => entry.scope == call.fullyQualifiedName}.toSet
+
+  private def callOutputEntries(call: Call): Set[SymbolStoreEntry] =
+    callEntries(call).filter {entry => entry.isOutput}
+
+  private def callInputEntries(call: Call): Set[SymbolStoreEntry] =
+    callEntries(call).filter {entry => entry.isInput}
+
   def locallyQualifiedInputs(call: Call): Map[String, WdlValue] = {
-    store.collect { case entry if entry.scope == call.fullyQualifiedName && entry.isInput =>
-      entry.key.name -> entry.wdlValue.get
+
+    def lookup(s: String): WdlValue = {
+      val workflow = call.parent.map {_.asInstanceOf[Workflow]} getOrElse {
+        throw new WdlExpressionException("Expecting 'call' to have a 'workflow' parent")
+      }
+      val matchedCall = workflow.calls.find {c => c.name == s}.getOrElse {
+        throw new WdlExpressionException(s"Expecting to find a call with name '$s'")
+      }
+      val callOutputs = callOutputEntries(matchedCall) map {entry =>
+        val value = entry.wdlValue match {
+          case Some(v) => v
+          case _ => throw new WdlExpressionException(s"Could not evaluate call '${matchedCall.name}', because '${entry.key.name}' is undefined")
+        }
+        entry.key.name -> value
+      }
+      WdlObject(callOutputs.toMap)
+    }
+
+    callInputEntries(call).map {entry =>
+      val value = entry.wdlValue match {
```

Comment:
It'd be nice to have a function which took an `entry` and returned a `value` so then it'd just look like:

    ...map {entry => entry.key.name -> doSomething(entry)}